### PR TITLE
add resource "azurerm_synapse_integration_runtime_self_hosted"

### PR DESCRIPTION
**Pull request by ms-henglu**
image

### CODE REVIEW

File: internal/services/synapse/synapse_integration_runtime_self_hosted_resource.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"auth_key_1": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"auth_key_2": {
```
Comment (katbyte) [inline internal/services/synapse/synapse_integration_runtime_self_hosted_resource.go:67]: should we make this a list ?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/synapse/synapse_integration_runtime_self_hosted_resource.go (hunk 1 above)
Comment (ms-henglu) [reply internal/services/synapse/synapse_integration_runtime_self_hosted_resource.go:67]: @katbyte , I think it's not necessary,  data_factory_integration_runtime_self_hosted is a very similar resource and it uses 2 properties, we can keep consistent with it.

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_integration_runtime_self_hosted#auth_key_1

Comment (katbyte) [reply internal/services/synapse/synapse_integration_runtime_self_hosted_resource.go:67]: ok so looking at the docs & given 1 & 2 are not descriptive, and it seems key 1 is primary, and 2 is secondary can we make this authorization 
```suggestion
			"authorization_key_primary": {
				Type:     pluginsdk.TypeString,
				Computed: true,
			},

			"authorization_key_secondary": {
```

and lets remove the abbreviation

datafactory should also be updated as 1 & 2 do no reflect what the keys are, could we do that/add a todo 3.0 there?

**Comment by ms-henglu** [comment]
Hi @katbyte , I've added a commit according to your suggestions. Please take another look, thanks!

**Review by katbyte — APPROVED** [review]
Thanks @ms-henglu - LGTM 🏗️

**Pull request merged**